src/EventManager.php

```diff
@@ -49,9 +52,17 @@ public function publish(string $event, array &$context)
             if (preg_match($s->getPattern(), $event)) {
                 $class = $s->getClass();
                 $this->g->getLogger()->addDebug("  Handling with {$class}.");
-                $c = $class::handleEvent($this->g, $event, $context);
+
+                $eventContext = new EventContext($event, $s->getPattern(), $contextData);
+
+                $returnedEventContext = $class::handleEvent($this->g, $eventContext);
+                if ($returnedEventContext->hasDataChanged($contextData)) {
```

Comment:
Why does this if matter? Why not just write over `$contextData` with `$returnedEventContext->getData()` no matter what?